chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,270 @@
+    try:
+        project = client.get_project(name=project_name)
+        return project.name.split("/")[-1]
+        # Extract number from 'projects/123456789'
+    except Exception as e:
+        logger.error(f"Error retrieving project number: {e}")
+        return None
+
+
+def apphub_app_list_services(project_id: str, location: str, application_id: str):
+    """Retrieves a list of service URIs associated with an AppHub application.
+
+    This function queries the AppHub API to list all services deployed within
+    a specific application and extracts their associated URIs. It replaces the
+    project number with the project ID in the URI for improved clarity.
+
+    Args:
+        project_id (str): The ID of the Google Cloud project containing the application.
+        location (str): The geographic location (e.g., "us-central1") of the application.
+        application_id (str): The unique identifier for the AppHub application.
+
+    Returns:
+        list: A list of strings, where each string is the URI of a service associated
+              with the specified application. The URIs are formatted to contain the
+              project ID instead of the project number."""
+    name_app = f"projects/{project_id}/locations/{location}/applications/{application_id}"
+    client = apphub_v1.AppHubClient()
+
+    request = apphub_v1.ListServicesRequest(parent=name_app)
+
+    service_uris = []
+    logger.info("Services for given Application in Apphub")
+    for response in client.list_services(request=request):
+        uri = response.service_reference.uri.replace(
+            get_project_number(project_id), project_id
+        )
+        updated_uri = (
+            uri[uri.find("/projects/") :] if "/projects/" in uri else uri
+        )
+        service_uris.append(updated_uri)
+    logger.info(updated_uri)
+    return service_uris
+
+
+def get_url_map_essentials(project_id: str):
+    """Retrieves essential information for URL maps in a Google Cloud project.
+
+    This function queries the Google Compute Engine API to fetch key details
+    about URL maps (including regional and global) within the specified project.
+    It yields a dictionary for each URL map, containing the following:
+
+    * region: The region (or "global") where the URL map is located.
+    * service: The default backend service associated with the URL map
+    * self_link: The selfLink URL uniquely identifying the URL map resource.
+    Args:
+        project_id (str): The ID of the Google Cloud project to query.
+
+    Yields:
+        dict: A dictionary containing the region, service, and self_link of each
+            URL map found in the project.
+
+    Raises:
+        Error: If there's an error communicating
+            with the Google Cloud API (e.g., authentication issues, invalid project ID).
+    """
+    client = compute_v1.UrlMapsClient()
+    request = compute_v1.AggregatedListUrlMapsRequest(project=project_id)
```

Comment:
I'm still confused why we cannot use the load_credentials/get_context functions as in other activities? Why is this different here?